data/pgroonga--3.2.5--3.2.4.sql

```diff
@@ -1 +1,5 @@
 -- Downgrade SQL
+
+DROP FUNCTION pgroonga_regexp_text_array;
+DROP OPERATOR FAMILY pgroonga_text_array_regexp_ops_v2 USING pgroonga;
+DROP OPERATOR CLASS pgroonga_text_array_regexp_ops_v2;
```

Comment:
この順番で実行して大丈夫？

依存されている方から順番に削除しないと削除できなくない？

ダウングレードのテストって用意済みだったっけ？

Reply:
ダウングレードのテストは用意できてないですね。
先にそっちを作ろうかと思います。

あと、↑の順番は確かにダメそうな気がするので、それを確認するためにもダウングレードのテストを先に作ります。

Reply:
`DROP FUNCCTION` -> `DROP OPERATOR CLASS` -> `DROP OPERATOR FAMILY`の順で削除するようにしました。

Reply:
あぁ、マージ前だったから確認できていないのか。

Reply:
以下のように失敗したので、CIで`data/`配下のダウングレード用SQLの動作を確認できていました。

```
  + psql downgrade -c 'ALTER EXTENSION pgroonga UPDATE TO '\''3.2.4'\'''
  2024-11-14 00:24:59.760 UTC [22262] ERROR:  cannot drop function pgroonga_regexp_text_array(text[],text) because other objects depend on it
  2024-11-14 00:24:59.760 UTC [22262] DETAIL:  operator &~(text[],text) depends on function pgroonga_regexp_text_array(text[],text)
  	operator class pgroonga_text_array_regexp_ops_v2 for access method pgroonga depends on operator &~(text[],text)
  2024-11-14 00:24:59.760 UTC [22262] HINT:  Use DROP ... CASCADE to drop the dependent objects too.
  2024-11-14 00:24:59.760 UTC [22262] STATEMENT:  ALTER EXTENSION pgroonga UPDATE TO '3.2.4'
  ERROR:  cannot drop function pgroonga_regexp_text_array(text[],text) because other objects depend on it
  DETAIL:  operator &~(text[],text) depends on function pgroonga_regexp_text_array(text[],text)
  operator class pgroonga_text_array_regexp_ops_v2 for access method pgroonga depends on operator &~(text[],text)
  HINT:  Use DROP ... CASCADE to drop the dependent objects too.
```

https://github.com/pgroonga/pgroonga/actions/runs/11827943749/job/32957452706#step:17:4689

Reply:
エラーメッセージによると、
operator は function に依存している。
operator class は operator に依存している。

function を drop しようとすると、 operator があってdropできない。
operatorをdropしようとすると、operator class があってdropできない。

なので、 operator class -> operator -> function の順でdropしないといけない。

Reply:
https://www.postgresql.jp/document/16/html/sql-dropopclass.html によると`DROP OPERATOR CLASS`は演算子族を削除しない（暗黙的に作られた演算子族であっても削除しない）ので、ダウングレード時は明示的に`DROP OPERATOR FAMILY`する必要がある。

Reply:
operator family -> operator class -> operator -> function の順で`DROP`するのが良さそう。